lib/pacemaker/pcmk_graph_producer.c

```diff
@@ -88,8 +88,13 @@ add_maintenance_nodes(xmlNode *xml, const pcmk_scheduler_t *scheduler)
          iter != NULL; iter = iter->next) {
         const pcmk_node_t *node = iter->data;
 
-        if (pcmk__is_pacemaker_remote_node(node) &&
-            (node->details->maintenance != node->details->remote_maintenance)) {
+        if (!pcmk__is_pacemaker_remote_node(node)) {
+            continue;
+        }
+        if ((node->details->maintenance
```

Comment:
could still use
```
node->details->maintenance
!= pcmk_is_set(node->private->flags, pcmk__node_remote_maint)
```

if desired. Matter of preference

Reply:
boolean truth is any nonzero not 1, better never to compare boolean equality unless you're really sure it can only be a particular value